core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -473,6 +473,16 @@ UNWIND(	.cantunwind)
 	bl	cpu_is_ready
 
 #if defined (CFG_BOOT_SECONDARY_REQUEST)
+spin:
+	/* spin until secondary boot request */
+	wfe
+	bl  get_core_pos
+	ldr r1, =spin_table
+	add r1, r1, r0, lsl #2	/* size of uint32_t */
+	ldr r0, [r1]
+	cmp	r0, #0
+	beq	spin
+
```

Comment:
not really, what if the NS entry is indeed at address 0x0 (who knows what hardware can come up with), so we cannot check `ns_entry_addrs` against 0 for that matter.